File: 02_algorithm/programmers/Level2/카펫.py
def solution(brown, yellow):
    x_value, y_value = yellow, 0
    while True:
        if not yellow % x_value:
            y_value = yellow // x_value
            if 2 * (x_value + y_value) + 4 == brown:
                return [x_value + 2, y_value + 2]
        x_value -= 1


print(solution(10, 2))
print(solution(8, 1))
print(solution(24, 24))

File: 06_js/04_algorithm/programmers/Level2/카펫.js
function solution(brown, yellow) {
  let x = yellow;
  let y = 0;
  while (true) {
    if (yellow % x === 0) {
      y = parseInt(yellow / x);
      if (2 * (x + y) + 4 === brown) {
        return [x + 2, y + 2];
      }
    }
    x -= 1;
  }
}

console.log(solution(10, 2)); // [4, 3]
console.log(solution(8, 1)); // [3, 3]
console.log(solution(24, 24)); // [8, 6]